Retries exhausted validating REST API via acceptance tests

**Detailed Description**
Acceptance tests failing in performance environment. Different REST API calls that validate the transaction sporadically exhaust all retries. When manually invoking those URLs later, they succeed. This indicates a timing issue that could be remedied by tweaking the retry counts, backoff delay, initial delay, etc. But unfortunately none of these are configurable.

**Actual Behavior**
```
 2020-11-23 21:20:19,296 DEBUG [main ] c.h.m.t.e.a.c.TokenClient Transferred 2350 tokens of 0.0.2661 from 0.0.90 to 0.0.2662
   And I transfer 2350 tokens to recipient                                                                     # com.hedera.mirror.test.e2e.acceptance.steps.TokenFeature.transferTokensToRecipient(int)
2020-11-23 21:20:19,298 DEBUG [main ] o.s.r.s.RetryTemplate Retry: count=0
 2020-11-23 21:20:19,299 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Verify balance for account '0.0.90' is returned by Mirror Node
 2020-11-23 21:20:19,306 DEBUG [reactor-http-epoll-1] r.n.h.c.HttpClientConnect [id: 0x46bc8a02, L:/10.44.0.174:58994 - R:34.67.159.125/34.67.159.125:80] Handler is being applied: {uri=http://34.67.159.125/api/v1/balances?account.id=0.0.90, method=GET}
 2020-11-23 21:20:19,322 DEBUG [reactor-http-epoll-1] r.n.h.c.HttpClientOperations [id: 0x46bc8a02, L:/10.44.0.174:58994 - R:34.67.159.125/34.67.159.125:80] Received response (auto-read:false) : [Access-Control-Allow-Origin=*, Content-Type=application/json; charset=utf-8, Content-Length=129, ETag=W/"81-v3w0XcrVPTKe0dYGbGHIJvKLV70", Vary=Accept-Encoding, Date=Mon, 23 Nov 2020 21:20:19 GMT, Connection=close]
 2020-11-23 21:20:19,323 DEBUG [reactor-http-epoll-1] r.n.h.c.HttpClientOperations [id: 0x46bc8a02, L:/10.44.0.174:58994 - R:34.67.159.125/34.67.159.125:80] Received last HTTP packet
 2020-11-23 21:20:19,324 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Endpoint /{endpoint}?{key}={accountId} returned 200 OK
 2020-11-23 21:20:24,347 DEBUG [main ] o.s.r.s.RetryTemplate Checking for rethrow: count=1
 2020-11-23 21:20:24,347 DEBUG [main ] o.s.r.s.RetryTemplate Retry: count=1
 2020-11-23 21:20:24,348 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Verify balance for account '0.0.90' is returned by Mirror Node
 2020-11-23 21:20:24,353 DEBUG [reactor-http-epoll-2] r.n.h.c.HttpClientConnect [id: 0x3e1555bb, L:/10.44.0.174:59126 - R:34.67.159.125/34.67.159.125:80] Handler is being applied: {uri=http://34.67.159.125/api/v1/balances?account.id=0.0.90, method=GET}
 2020-11-23 21:20:24,367 DEBUG [reactor-http-epoll-2] r.n.h.c.HttpClientOperations [id: 0x3e1555bb, L:/10.44.0.174:59126 - R:34.67.159.125/34.67.159.125:80] Received response (auto-read:false) : [Access-Control-Allow-Origin=*, Content-Type=application/json; charset=utf-8, Content-Length=129, ETag=W/"81-v3w0XcrVPTKe0dYGbGHIJvKLV70", Vary=Accept-Encoding, Date=Mon, 23 Nov 2020 21:20:24 GMT, Connection=close]
 2020-11-23 21:20:24,368 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Endpoint /{endpoint}?{key}={accountId} returned 200 OK
 2020-11-23 21:20:24,368 DEBUG [reactor-http-epoll-2] r.n.h.c.HttpClientOperations [id: 0x3e1555bb, L:/10.44.0.174:59126 - R:34.67.159.125/34.67.159.125:80] Received last HTTP packet
 2020-11-23 21:20:29,370 DEBUG [main ] o.s.r.s.RetryTemplate Checking for rethrow: count=2
 2020-11-23 21:20:29,371 DEBUG [main ] o.s.r.s.RetryTemplate Retry: count=2
 2020-11-23 21:20:29,371 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Verify balance for account '0.0.90' is returned by Mirror Node
 2020-11-23 21:20:29,375 DEBUG [reactor-http-epoll-3] r.n.h.c.HttpClientConnect [id: 0xa857ab25, L:/10.44.0.174:59256 - R:34.67.159.125/34.67.159.125:80] Handler is being applied: {uri=http://34.67.159.125/api/v1/balances?account.id=0.0.90, method=GET}
 2020-11-23 21:20:29,390 DEBUG [reactor-http-epoll-3] r.n.h.c.HttpClientOperations [id: 0xa857ab25, L:/10.44.0.174:59256 - R:34.67.159.125/34.67.159.125:80] Received response (auto-read:false) : [Access-Control-Allow-Origin=*, Content-Type=application/json; charset=utf-8, Content-Length=129, ETag=W/"81-v3w0XcrVPTKe0dYGbGHIJvKLV70", Vary=Accept-Encoding, Date=Mon, 23 Nov 2020 21:20:29 GMT, Connection=close]
 2020-11-23 21:20:29,391 DEBUG [reactor-http-epoll-3] r.n.h.c.HttpClientOperations [id: 0xa857ab25, L:/10.44.0.174:59256 - R:34.67.159.125/34.67.159.125:80] Received last HTTP packet
 2020-11-23 21:20:29,391 DEBUG [main ] c.h.m.t.e.a.c.MirrorNodeClient Endpoint /{endpoint}?{key}={accountId} returned 200 OK
 2020-11-23 21:20:29,393 DEBUG [main ] o.s.r.s.RetryTemplate Checking for rethrow: count=3
 2020-11-23 21:20:29,393 DEBUG [main ] o.s.r.s.RetryTemplate Retry failed last attempt: count=3
 2020-11-23 21:20:29,394 ERROR [main ] c.h.m.t.e.a.s.TokenFeature Received 200 from REST API, failed verification after 5 retries w: 
Expecting actual not to be empty
   Then the mirror node REST API should return status 200 for token fund flow                                  # com.hedera.mirror.test.e2e.acceptance.steps.TokenFeature.verifyMirrorTokenFundFlow(int)
      java.lang.AssertionError: 
Expecting actual not to be empty
	at com.hedera.mirror.test.e2e.acceptance.steps.TokenFeature.verifyBalances(TokenFeature.java:381)
	at com.hedera.mirror.test.e2e.acceptance.steps.TokenFeature.verifyMirrorTokenFundFlow(TokenFeature.java:286)
	at ???.the mirror node REST API should return status 200 for token fund flow(file:///home/jenkins/agent/workspace/hcs-acceptance/hedera-mirror-test/src/test/resources/features/hts/hts.feature:33)
```

**Expected Behavior**
No 404 not found and acceptance tests pass

**Environment:**
- Version: 0.23.1

**Additional Context**
Add any other context about the problem here. Attach any logs here, if applicable.

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/handler/SDKClientHandler.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/handler/SDKClientHandler.java
@@ -100,7 +100,7 @@ public class SDKClientHandler {
         return topicId;
     }
 
-    public List<TransactionId> submitTopicMessage(ConsensusTopicId topicId, String message) throws HederaStatusException {
+    public List<TransactionId> submitTopicMessage(ConsensusTopicId topicId, byte[] message) throws HederaStatusException {
         ConsensusMessageSubmitTransaction consensusMessageSubmitTransaction = new ConsensusMessageSubmitTransaction()
                 .setTopicId(topicId)
                 .setMessage(message);

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
@@ -51,7 +51,7 @@ public class TopicMessagePublishRequest {
 
     public byte[] getMessage() {
         if (randomBytes == null) {
-            randomBytes = new byte[messageByteSize > 8 ? messageByteSize - Long.BYTES : 0];
+            randomBytes = new byte[messageByteSize > Long.BYTES ? messageByteSize - Long.BYTES : 0];
             new SecureRandom().nextBytes(randomBytes);
         }
 

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
@@ -21,13 +21,12 @@ package com.hedera.mirror.grpc.jmeter.props;
  */
 
 import com.google.common.primitives.Longs;
-import java.time.Instant;
+import java.security.SecureRandom;
 import lombok.Builder;
 import lombok.Data;
 import lombok.ToString;
 import lombok.extern.log4j.Log4j2;
-import org.apache.commons.codec.binary.Base64;
-import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.ArrayUtils;
 
 import com.hedera.hashgraph.sdk.account.AccountId;
 import com.hedera.hashgraph.sdk.consensus.ConsensusTopicId;
@@ -48,18 +47,15 @@ public class TopicMessagePublishRequest {
     @ToString.Exclude
     private Ed25519PrivateKey operatorPrivateKey;
 
-    private String randomAlphanumeric;
+    private byte[] randomAlphanumeric;
 
-    public String getMessage() {
-        int timeStampBytes = 8;
+    public byte[] getMessage() {
         if (randomAlphanumeric == null) {
-            int additionalBytes = messageByteSize <= timeStampBytes ? 0 : messageByteSize - 8;
-            randomAlphanumeric = RandomStringUtils.randomAlphanumeric(additionalBytes);
+            randomAlphanumeric = new byte[messageByteSize > 8 ? messageByteSize - Long.BYTES : 0];
+            new SecureRandom().nextBytes(randomAlphanumeric);
         }
 
-        // set current time stamp to first 8 bytes of message
-        byte[] timeRefBytes = Longs.toByteArray(Instant.now().toEpochMilli());
-
-        return Base64.encodeBase64String(timeRefBytes) + randomAlphanumeric;
+        // set current time stamp to first 8 bytes of message and random message after
+        return ArrayUtils.addAll(Longs.toByteArray(System.currentTimeMillis()), randomAlphanumeric);
     }
 }

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/props/TopicMessagePublishRequest.java
@@ -47,15 +47,15 @@ public class TopicMessagePublishRequest {
     @ToString.Exclude
     private Ed25519PrivateKey operatorPrivateKey;
 
-    private byte[] randomAlphanumeric;
+    private byte[] randomBytes;
 
     public byte[] getMessage() {
-        if (randomAlphanumeric == null) {
-            randomAlphanumeric = new byte[messageByteSize > 8 ? messageByteSize - Long.BYTES : 0];
-            new SecureRandom().nextBytes(randomAlphanumeric);
+        if (randomBytes == null) {
+            randomBytes = new byte[messageByteSize > 8 ? messageByteSize - Long.BYTES : 0];
+            new SecureRandom().nextBytes(randomBytes);
         }
 
-        // set current time stamp to first 8 bytes of message and random message after
-        return ArrayUtils.addAll(Longs.toByteArray(System.currentTimeMillis()), randomAlphanumeric);
+        // set current time stamp to first 8 bytes of message and random bytes message after
+        return ArrayUtils.addAll(Longs.toByteArray(System.currentTimeMillis()), randomBytes);
     }
 }

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/sampler/result/HCSSamplerResult.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/sampler/result/HCSSamplerResult.java
@@ -23,12 +23,10 @@ package com.hedera.mirror.grpc.jmeter.sampler.result;
 import com.google.common.base.Stopwatch;
 import com.google.common.primitives.Longs;
 import java.time.Instant;
-import java.util.Arrays;
 import java.util.concurrent.TimeUnit;
 import lombok.Data;
 import lombok.experimental.SuperBuilder;
 import lombok.extern.log4j.Log4j2;
-import org.apache.commons.codec.binary.Base64;
 import org.apache.commons.math3.stat.descriptive.SummaryStatistics;
 
 @SuperBuilder
@@ -163,16 +161,10 @@ public abstract class HCSSamplerResult<T> {
 
         byte[] message = getMessageByteArray(currentResponse);
         try {
-            byte[] decodedMessage = Base64.decodeBase64(message);
-            publishInstant = retrieveInstantFromArray(decodedMessage);
+            publishInstant = retrieveInstantFromArray(message);
             if (isInstantOutOfRange(publishInstant)) {
-                publishInstant = retrieveInstantFromArray(message);
-
-                // support non encoded version
-                if (isInstantOutOfRange(publishInstant)) {
-                    log.debug("publishInstant is out of range: {}", publishInstant);
-                    publishInstant = null;
-                }
+                log.debug("publishInstant is out of range: {}", publishInstant);
+                publishInstant = null;
             }
         } catch (Exception ex) {
             log.debug("response message contains invalid publish millisecond value: {}", message);
@@ -183,7 +175,7 @@ public abstract class HCSSamplerResult<T> {
     }
 
     private Instant retrieveInstantFromArray(byte[] message) {
-        Long publishMillis = Longs.fromByteArray(Arrays.copyOfRange(message, 0, 8));
+        Long publishMillis = Longs.fromByteArray(message);
         return Instant.ofEpochMilli(publishMillis);
     }
 

--- a/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/sampler/result/HCSSamplerResult.java
+++ b/hedera-mirror-grpc/src/test/java/com/hedera/mirror/grpc/jmeter/sampler/result/HCSSamplerResult.java
@@ -22,6 +22,7 @@ package com.hedera.mirror.grpc.jmeter.sampler.result;
 
 import com.google.common.base.Stopwatch;
 import com.google.common.primitives.Longs;
+import java.nio.charset.StandardCharsets;
 import java.time.Instant;
 import java.util.concurrent.TimeUnit;
 import lombok.Data;
@@ -167,7 +168,8 @@ public abstract class HCSSamplerResult<T> {
                 publishInstant = null;
             }
         } catch (Exception ex) {
-            log.debug("response message contains invalid publish millisecond value: {}", message);
+            log.debug("response message contains invalid publish millisecond value: '{}', ex: {}",
+                    new String(message, StandardCharsets.UTF_8), ex.getMessage());
             publishInstant = null;
         }
 
@@ -175,6 +177,10 @@ public abstract class HCSSamplerResult<T> {
     }
 
     private Instant retrieveInstantFromArray(byte[] message) {
+        if (message == null || message.length < Long.BYTES) {
+            return Instant.MAX;
+        }
+
         Long publishMillis = Longs.fromByteArray(message);
         return Instant.ofEpochMilli(publishMillis);
     }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/SubscriptionResponse.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/SubscriptionResponse.java
@@ -28,7 +28,6 @@ import java.util.ArrayList;
 import java.util.List;
 import lombok.Data;
 import lombok.extern.log4j.Log4j2;
-import org.apache.commons.codec.binary.Base64;
 
 import com.hedera.hashgraph.sdk.mirror.MirrorConsensusTopicResponse;
 import com.hedera.hashgraph.sdk.mirror.MirrorSubscriptionHandle;
@@ -65,8 +64,7 @@ public class SubscriptionResponse {
             MirrorConsensusTopicResponse mirrorConsensusTopicResponse = mirrorHCSResponseResponse
                     .getMirrorConsensusTopicResponse();
 
-            Instant publishInstant = Instant
-                    .ofEpochMilli(Longs.fromByteArray(Base64.decodeBase64(mirrorConsensusTopicResponse.message)));
+            Instant publishInstant = Instant.ofEpochMilli(Longs.fromByteArray(mirrorConsensusTopicResponse.message));
 
             long publishSeconds = publishInstant.getEpochSecond();
             long consensusSeconds = mirrorConsensusTopicResponse.consensusTimestamp.getEpochSecond();

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
@@ -30,7 +30,7 @@ import java.util.List;
 import java.util.Map;
 import lombok.Value;
 import lombok.extern.log4j.Log4j2;
-import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.ArrayUtils;
 
 import com.hedera.hashgraph.sdk.HederaStatusException;
 import com.hedera.hashgraph.sdk.TransactionId;
@@ -117,8 +117,9 @@ public class TopicClient extends AbstractNetworkClient {
         log.debug("Publishing {} message(s) to topicId : {}.", numMessages, topicId);
         List<TransactionReceipt> transactionReceiptList = new ArrayList<>();
         for (int i = 0; i < numMessages; i++) {
-            byte[] byteArray = Longs.toByteArray(Instant.now().toEpochMilli());
-            String message = Base64.encodeBase64String(byteArray) + "_" + baseMessage + "_" + i + 1;
+            byte[] publishTimestampByteArray = Longs.toByteArray(System.currentTimeMillis());
+            byte[] suffixByteArray = ("_" + baseMessage + "_" + i + 1).getBytes();
+            byte[] message = ArrayUtils.addAll(publishTimestampByteArray, suffixByteArray);
 
             if (verify) {
                 transactionReceiptList.addAll(publishMessageToTopicAndVerify(topicId, message, submitKey));
@@ -130,7 +131,7 @@ public class TopicClient extends AbstractNetworkClient {
         return transactionReceiptList;
     }
 
-    public List<TransactionId> publishMessageToTopic(ConsensusTopicId topicId, String message,
+    public List<TransactionId> publishMessageToTopic(ConsensusTopicId topicId, byte[] message,
                                                      Ed25519PrivateKey submitKey) throws HederaStatusException {
         ConsensusMessageSubmitTransaction consensusMessageSubmitTransaction = new ConsensusMessageSubmitTransaction()
                 .setTopicId(topicId)
@@ -144,13 +145,13 @@ public class TopicClient extends AbstractNetworkClient {
             recordPublishInstants.put(0L, transactionRecord.consensusTimestamp);
         }
 
-        log.trace("Published message : '{}' to topicId : {} with consensusTimestamp: {}", message, topicId,
+        log.trace("Published message : '{}' to topicId : {} with consensusTimestamp: {}", new String(message), topicId,
                 transactionRecord.consensusTimestamp);
 
         return transactionIdList;
     }
 
-    public List<TransactionReceipt> publishMessageToTopicAndVerify(ConsensusTopicId topicId, String message,
+    public List<TransactionReceipt> publishMessageToTopicAndVerify(ConsensusTopicId topicId, byte[] message,
                                                                    Ed25519PrivateKey submitKey) throws HederaStatusException {
         List<TransactionId> transactionIdList = publishMessageToTopic(topicId, message, submitKey);
         List<TransactionReceipt> transactionReceipts = new ArrayList<>();

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
@@ -21,6 +21,7 @@ package com.hedera.mirror.test.e2e.acceptance.client;
  */
 
 import com.google.common.primitives.Longs;
+import java.nio.charset.StandardCharsets;
 import java.time.Duration;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
@@ -111,14 +112,13 @@ public class TopicClient extends AbstractNetworkClient {
     }
 
     public List<TransactionReceipt> publishMessagesToTopic(ConsensusTopicId topicId, String baseMessage,
-                                                           Ed25519PrivateKey submitKey,
-                                                           int numMessages, boolean verify) throws HederaStatusException
-            , InterruptedException {
+                                                           Ed25519PrivateKey submitKey, int numMessages,
+                                                           boolean verify) throws HederaStatusException {
         log.debug("Publishing {} message(s) to topicId : {}.", numMessages, topicId);
         List<TransactionReceipt> transactionReceiptList = new ArrayList<>();
         for (int i = 0; i < numMessages; i++) {
             byte[] publishTimestampByteArray = Longs.toByteArray(System.currentTimeMillis());
-            byte[] suffixByteArray = ("_" + baseMessage + "_" + (i + 1)).getBytes();
+            byte[] suffixByteArray = ("_" + baseMessage + "_" + (i + 1)).getBytes(StandardCharsets.UTF_8);
             byte[] message = ArrayUtils.addAll(publishTimestampByteArray, suffixByteArray);
 
             if (verify) {
@@ -145,8 +145,10 @@ public class TopicClient extends AbstractNetworkClient {
             recordPublishInstants.put(0L, transactionRecord.consensusTimestamp);
         }
 
-        log.trace("Published message : '{}' to topicId : {} with consensusTimestamp: {}", new String(message), topicId,
-                transactionRecord.consensusTimestamp);
+        if (log.isTraceEnabled()) {
+            log.trace("Published message : '{}' to topicId : {} with consensusTimestamp: {}",
+                    new String(message, StandardCharsets.UTF_8), topicId, transactionRecord.consensusTimestamp);
+        }
 
         return transactionIdList;
     }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TopicClient.java
@@ -118,7 +118,7 @@ public class TopicClient extends AbstractNetworkClient {
         List<TransactionReceipt> transactionReceiptList = new ArrayList<>();
         for (int i = 0; i < numMessages; i++) {
             byte[] publishTimestampByteArray = Longs.toByteArray(System.currentTimeMillis());
-            byte[] suffixByteArray = ("_" + baseMessage + "_" + i + 1).getBytes();
+            byte[] suffixByteArray = ("_" + baseMessage + "_" + (i + 1)).getBytes();
             byte[] message = ArrayUtils.addAll(publishTimestampByteArray, suffixByteArray);
 
             if (verify) {

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/AcceptanceTestProperties.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/AcceptanceTestProperties.java
@@ -22,6 +22,7 @@ package com.hedera.mirror.test.e2e.acceptance.config;
 
 import java.time.Duration;
 import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
 import lombok.Data;
@@ -58,4 +59,11 @@ public class AcceptanceTestProperties {
 
     @NotNull
     private Duration subscribeRetryBackoffPeriod = Duration.ofMillis(5000);
+
+    @Min(1)
+    @Max(60)
+    private int restPollRetries = 60;
+
+    @NotNull
+    private Duration restRetryBackoffPeriod = Duration.ofMillis(1000);
 }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/ClientConfiguration.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/ClientConfiguration.java
@@ -125,7 +125,7 @@ public class ClientConfiguration {
         Jackson2JsonEncoder jackson2JsonEncoder = new Jackson2JsonEncoder(objectMapper, MediaType.APPLICATION_JSON);
 
         return WebClient.builder()
-                .baseUrl(acceptanceTestProperties.getMirrorRestAddress())
+                .baseUrl(acceptanceTestProperties.getRestPollingProperties().getBaseUrl())
                 .clientConnector(new ReactorClientHttpConnector(HttpClient.from(tcpClient)))
                 .codecs(clientCodecConfigurer -> {
                     clientCodecConfigurer.defaultCodecs().jackson2JsonDecoder(jackson2JsonDecoder);

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/RestPollingProperties.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/RestPollingProperties.java
@@ -26,6 +26,8 @@ import javax.validation.constraints.Min;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
 import lombok.Data;
+import org.hibernate.validator.constraints.time.DurationMax;
+import org.hibernate.validator.constraints.time.DurationMin;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.stereotype.Component;
 import org.springframework.validation.annotation.Validated;
@@ -39,6 +41,8 @@ public class RestPollingProperties {
     @NotBlank
     private String baseUrl;
 
+    @DurationMin(seconds = 0L)
+    @DurationMax(seconds = 10L)
     @NotNull
     private Duration delay = Duration.ofMillis(1000);
 

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/RestPollingProperties.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/config/RestPollingProperties.java
@@ -22,6 +22,7 @@ package com.hedera.mirror.test.e2e.acceptance.config;
 
 import java.time.Duration;
 import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
 import lombok.Data;
@@ -30,32 +31,18 @@ import org.springframework.stereotype.Component;
 import org.springframework.validation.annotation.Validated;
 
 @Component
-@ConfigurationProperties(prefix = "hedera.mirror.test.acceptance")
+@ConfigurationProperties(prefix = "hedera.mirror.test.acceptance.rest")
 @Data
 @Validated
-public class AcceptanceTestProperties {
-    private final RestPollingProperties restPollingProperties;
+public class RestPollingProperties {
 
     @NotBlank
-    private String nodeAddress;
-    @NotBlank
-    private String nodeId;
-    @NotBlank
-    private String mirrorNodeAddress;
-    @NotBlank
-    private String operatorId;
-    @NotBlank
-    private String operatorKey;
-    @NotNull
-    private Duration messageTimeout = Duration.ofSeconds(20);
-    @NotNull
-    private Long existingTopicNum;
-
-    private boolean emitBackgroundMessages = false;
-
-    @Max(5)
-    private int subscribeRetries = 5;
+    private String baseUrl;
 
     @NotNull
-    private Duration subscribeRetryBackoffPeriod = Duration.ofMillis(5000);
+    private Duration delay = Duration.ofMillis(1000);
+
+    @Min(1)
+    @Max(60)
+    private int maxAttempts = 60;
 }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
@@ -272,16 +272,16 @@ public class TokenFeature {
 
     @Then("the mirror node REST API should return status {int}")
     @Retryable(value = {AssertionError.class, AssertionFailedError.class},
-            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
-            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
+            backoff = @Backoff(delayExpression = "#{@restPollingProperties.delay.toMillis()}"),
+            maxAttemptsExpression = "#{@restPollingProperties.maxAttempts}")
     public void verifyMirrorAPIResponses(int status) {
         verifyTransactions(status);
     }
 
     @Then("the mirror node REST API should return status {int} for token fund flow")
     @Retryable(value = {AssertionError.class, AssertionFailedError.class},
-            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
-            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
+            backoff = @Backoff(delayExpression = "#{@restPollingProperties.delay.toMillis()}"),
+            maxAttemptsExpression = "#{@restPollingProperties.maxAttempts}")
     public void verifyMirrorTokenFundFlow(int status) {
         verifyBalances();
         verifyTransactions(status);
@@ -291,16 +291,16 @@ public class TokenFeature {
 
     @Then("the mirror node REST API should confirm token update")
     @Retryable(value = {AssertionError.class, AssertionFailedError.class},
-            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
-            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
+            backoff = @Backoff(delayExpression = "#{@restPollingProperties.delay.toMillis()}"),
+            maxAttemptsExpression = "#{@restPollingProperties.maxAttempts}")
     public void verifyMirrorTokenUpdateFlow() {
         verifyTokenUpdate();
     }
 
     @Then("the mirror node REST API should return status {int} for transaction {string}")
     @Retryable(value = {AssertionError.class, AssertionFailedError.class},
-            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
-            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
+            backoff = @Backoff(delayExpression = "#{@restPollingProperties.delay.toMillis()}"),
+            maxAttemptsExpression = "#{@restPollingProperties.maxAttempts}")
     public void verifyMirrorRestTransactionIsPresent(int status, String transactionIdString) {
         MirrorTransactionsResponse mirrorTransactionsResponse = mirrorClient.getTransactions(transactionIdString);
 
@@ -466,7 +466,7 @@ public class TokenFeature {
     @Recover
     public void recover(AssertionError t) {
         log.error("REST API response verification failed after {} retries w: {}",
-                acceptanceProps.getRestPollRetries(), t.getMessage());
+                acceptanceProps.getRestPollingProperties().getMaxAttempts(), t.getMessage());
         throw t;
     }
 }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java
@@ -191,15 +191,11 @@ public class TokenFeature {
     }
 
     @Then("I transfer {int} tokens to recipient")
-    @Retryable(value = {StatusRuntimeException.class}, exceptionExpression = "#{message.contains('UNAVAILABLE') || " +
-            "message.contains('RESOURCE_EXHAUSTED')}")
     public void transferTokensToRecipient(int amount) throws HederaStatusException {
         transferTokens(tokenId, amount, tokenClient.getSdkClient().getOperatorId(), recipient.getAccountId());
     }
 
     @Then("I transfer {int} tokens of {int} to {int}")
-    @Retryable(value = {StatusRuntimeException.class}, exceptionExpression = "#{message.contains('UNAVAILABLE') || " +
-            "message.contains('RESOURCE_EXHAUSTED')}")
     public void transferTokens(int amount, int token, int recipient) throws HederaStatusException {
         transferTokens(new TokenId(token), amount, tokenClient.getSdkClient()
                 .getOperatorId(), new AccountId(recipient));
@@ -275,13 +271,17 @@ public class TokenFeature {
     }
 
     @Then("the mirror node REST API should return status {int}")
-    @Retryable(value = {AssertionError.class, AssertionFailedError.class}, backoff = @Backoff(delay = 5000))
+    @Retryable(value = {AssertionError.class, AssertionFailedError.class},
+            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
+            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
     public void verifyMirrorAPIResponses(int status) {
         verifyTransactions(status);
     }
 
     @Then("the mirror node REST API should return status {int} for token fund flow")
-    @Retryable(value = {AssertionError.class, AssertionFailedError.class}, backoff = @Backoff(delay = 5000))
+    @Retryable(value = {AssertionError.class, AssertionFailedError.class},
+            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
+            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
     public void verifyMirrorTokenFundFlow(int status) {
         verifyBalances();
         verifyTransactions(status);
@@ -290,13 +290,17 @@ public class TokenFeature {
     }
 
     @Then("the mirror node REST API should confirm token update")
-    @Retryable(value = {AssertionError.class, AssertionFailedError.class}, backoff = @Backoff(delay = 5000))
+    @Retryable(value = {AssertionError.class, AssertionFailedError.class},
+            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
+            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
     public void verifyMirrorTokenUpdateFlow() {
         verifyTokenUpdate();
     }
 
     @Then("the mirror node REST API should return status {int} for transaction {string}")
-    @Retryable(value = {AssertionError.class, AssertionFailedError.class}, backoff = @Backoff(delay = 5000))
+    @Retryable(value = {AssertionError.class, AssertionFailedError.class},
+            backoff = @Backoff(delayExpression = "#{@acceptanceTestProperties.restRetryBackoffPeriod.toMillis()}"),
+            maxAttemptsExpression = "#{@acceptanceTestProperties.restPollRetries}")
     public void verifyMirrorRestTransactionIsPresent(int status, String transactionIdString) {
         MirrorTransactionsResponse mirrorTransactionsResponse = mirrorClient.getTransactions(transactionIdString);
 
@@ -442,15 +446,14 @@ public class TokenFeature {
     }
 
     /**
-     * Recover method for REST verify operations. Method parameters of retry method must match this method after
-     * exception parameter
+     * Recover method for token transaction retry operations. Method parameters of retry method must match this method
+     * after exception parameter
      *
      * @param t
      */
     @Recover
-    public void recover(AssertionError t, int status) {
-        log.error("Received {} from REST API, failed verification after {} retries w: {}",
-                status, acceptanceProps.getSubscribeRetries(), t.getMessage());
+    public void recover(StatusRuntimeException t) {
+        log.error("Transaction submissions for token transaction failed after retries w: {}", t.getMessage());
         throw t;
     }
 
@@ -461,9 +464,9 @@ public class TokenFeature {
      * @param t
      */
     @Recover
-    public void recover(AssertionError t, String endpoint, int status) {
-        log.error("Received {} response from {}. Failed verification after {} retries w:" +
-                " {}", status, endpoint, acceptanceProps.getSubscribeRetries(), t.getMessage());
+    public void recover(AssertionError t) {
+        log.error("REST API response verification failed after {} retries w: {}",
+                acceptanceProps.getRestPollRetries(), t.getMessage());
         throw t;
     }
 }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TopicFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TopicFeature.java
@@ -28,6 +28,7 @@ import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
 import io.cucumber.junit.platform.engine.Cucumber;
 import io.grpc.StatusRuntimeException;
+import java.nio.charset.StandardCharsets;
 import java.time.Instant;
 import java.util.List;
 import java.util.concurrent.Executors;
@@ -364,7 +365,10 @@ public class TopicFeature {
             scheduler = Executors.newSingleThreadScheduledExecutor();
             scheduler.scheduleAtFixedRate(() -> {
                 try {
-                    topicClient.publishMessageToTopic(consensusTopicId, "backgroundMessage".getBytes(), submitKey);
+                    topicClient.publishMessageToTopic(
+                            consensusTopicId,
+                            "backgroundMessage".getBytes(StandardCharsets.UTF_8),
+                            submitKey);
                 } catch (HederaStatusException e) {
                     e.printStackTrace();
                 }

--- a/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TopicFeature.java
+++ b/hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TopicFeature.java
@@ -364,7 +364,7 @@ public class TopicFeature {
             scheduler = Executors.newSingleThreadScheduledExecutor();
             scheduler.scheduleAtFixedRate(() -> {
                 try {
-                    topicClient.publishMessageToTopic(consensusTopicId, "backgroundMessage", submitKey);
+                    topicClient.publishMessageToTopic(consensusTopicId, "backgroundMessage".getBytes(), submitKey);
                 } catch (HederaStatusException e) {
                     e.printStackTrace();
                 }

--- a/hedera-mirror-test/src/test/resources/application-default.yml
+++ b/hedera-mirror-test/src/test/resources/application-default.yml
@@ -7,10 +7,11 @@ hedera:
         messageTimeout: 20s
         # grpc endpoint
         mirrorNodeAddress: hcs.testnet.mirrornode.hedera.com:5600
-        # REST endpoint
-        mirrorRestAddress: http://testnet.mirrornode.hedera.com/api/v1
         nodeAddress: testnet
         nodeId: 0.0.3
         # Do not use use 0.0.2 or 0.0.50 for operator to ensure crypto transfers are not waived
         operatorId:
         operatorKey:
+        rest:
+          # REST endpoint
+          baseUrl: http://testnet.mirrornode.hedera.com/api/v1

